filters/builtin/header_test.go

```diff
@@ -186,64 +217,125 @@ func TestHeader(t *testing.T) {
 		args:       []interface{}{"Host", "www.example.org"},
 		valid:      true,
 		host:       "www.example.org",
+	}, {
+		msg:            "set request header from context",
+		filterName:     "setContextRequestHeader",
+		args:           []interface{}{"X-Test-Foo", "foo"},
+		context:        map[string]interface{}{"foo": "bar"},
+		valid:          true,
+		expectedHeader: http.Header{"X-Test-Request-Foo": []string{"bar"}},
+	}, {
+		msg:            "append request header from context",
+		filterName:     "appendContextRequestHeader",
+		args:           []interface{}{"X-Test-Foo", "foo"},
+		context:        map[string]interface{}{"foo": "baz"},
+		valid:          true,
+		requestHeader:  http.Header{"X-Test-Foo": []string{"bar"}},
+		expectedHeader: http.Header{"X-Test-Request-Foo": []string{"bar", "baz"}},
+	}, {
+		msg:        "set request host header from context",
+		filterName: "setContextRequestHeader",
+		args:       []interface{}{"Host", "foo"},
+		context:    map[string]interface{}{"foo": "www.example.org"},
+		valid:      true,
+		host:       "www.example.org",
+	}, {
+		msg:        "append request host header from context",
+		filterName: "appendContextRequestHeader",
+		args:       []interface{}{"Host", "foo"},
+		context:    map[string]interface{}{"foo": "www.example.org"},
+		valid:      true,
+		host:       "www.example.org",
+	}, {
+		msg:            "set response header from context",
+		filterName:     "setContextResponseHeader",
+		args:           []interface{}{"X-Test-Foo", "foo"},
+		context:        map[string]interface{}{"foo": "bar"},
+		valid:          true,
+		expectedHeader: http.Header{"X-Test-Foo": []string{"bar"}},
+	}, {
+		msg:            "append response header from context",
+		filterName:     "appendContextResponseHeader",
+		args:           []interface{}{"X-Test-Foo", "foo"},
+		context:        map[string]interface{}{"foo": "baz"},
+		valid:          true,
+		responseHeader: http.Header{"X-Test-Foo": []string{"bar"}},
+		expectedHeader: http.Header{"X-Test-Foo": []string{"bar", "baz"}},
```

Comment:
Nit I know this is test code, but it would be easier in my opinion if the tests were separated by filters.

Reply:
:+1: 

gropued these tests by the different filters